src/Microsoft.DotNet.Build.Tasks/AddDependenciesToProjectJson.cs

```diff
@@ -198,15 +195,44 @@ private void AddPackageItemsToDictionary(ref Dictionary<string, PackageItem> pac
             }
         }
 
-        // A versions file is of the form https://github.com/dotnet/versions/blob/master/build-info/dotnet/corefx/master/Latest_Packages.txt
-        private Dictionary<string, PackageItem> GatherPackageInformationFromVersionsFile(string versionsFile, VersionComparer comparer = null)
+        private static async System.Threading.Tasks.Task<Stream> GetStream(Uri uri)
+        {
+            if(uri.Scheme == "file")
+            {
+                return new FileStream(uri.LocalPath, FileMode.Open, FileAccess.Read, FileShare.Read);
+            }
+            else
+            {
+                return await new HttpClient().GetStreamAsync(uri);
+            }
+        }
+
+        private static async System.Threading.Tasks.Task<string> GetFileAsync(Uri uri)
+        {
+            using (var stream = GetStream(uri).Result)
+            using (StreamReader reader = new StreamReader(stream))
+            {
+                return await reader.ReadToEndAsync();
+            }
+        }
+
+        // A versions file is of the form https://raw.githubusercontent.com/dotnet/versions/master/build-info/dotnet/corefx/release/1.0.0/LKG_Packages.txt
+        private Dictionary<string, PackageItem> GatherPackageInformationFromVersionsFile(Uri uri, VersionComparer comparer = null)
         {
             Dictionary<string, PackageItem> packageItems = new Dictionary<string, PackageItem>();
-            if (!File.Exists(versionsFile))
+
+            string contents = string.Empty;
+            try
+            {
+                contents = GetFileAsync(uri).Result;
+            }
+            catch (AggregateException ae)
             {
-                Log.LogError("Specified versions file ({0}) does not exist.", versionsFile);
+                Log.LogError("Error: Unable to open '{0}', either the file does not exist locally or there is a network issue accessing that URI.", uri.ToString());
+                throw ae;
```

Comment:
If you are going to re-throw you should just do `throw;` so the original exception, including its original callstack, is rethrown.